src/pocketmine/block/Block.php

```diff
@@ -808,85 +842,84 @@ protected function recalculateBoundingBox(){
+				$v2 = null;
+			}
 
-		if($v2 !== null and !$bb->isVectorInYZ($v2)){
-			$v2 = null;
-		}
+			if($v3 !== null and !$bb->isVectorInXZ($v3)){
+				$v3 = null;
+			}
 
-		if($v3 !== null and !$bb->isVectorInXZ($v3)){
-			$v3 = null;
-		}
+			if($v4 !== null and !$bb->isVectorInXZ($v4)){
+				$v4 = null;
+			}
 
-		if($v4 !== null and !$bb->isVectorInXZ($v4)){
-			$v4 = null;
-		}
+			if($v5 !== null and !$bb->isVectorInXY($v5)){
+				$v5 = null;
+			}
 
-		if($v5 !== null and !$bb->isVectorInXY($v5)){
-			$v5 = null;
-		}
+			if($v6 !== null and !$bb->isVectorInXY($v6)){
+				$v6 = null;
+			}
 
-		if($v6 !== null and !$bb->isVectorInXY($v6)){
-			$v6 = null;
-		}
+			$vector = $v1;
 
-		$vector = $v1;
+			if($v2 !== null and ($vector === null or $pos1->distanceSquared($v2) < $pos1->distanceSquared($vector))){
+				$vector = $v2;
+			}
 
-		if($v2 !== null and ($vector === null or $pos1->distanceSquared($v2) < $pos1->distanceSquared($vector))){
-			$vector = $v2;
-		}
+			if($v3 !== null and ($vector === null or $pos1->distanceSquared($v3) < $pos1->distanceSquared($vector))){
+				$vector = $v3;
+			}
 
-		if($v3 !== null and ($vector === null or $pos1->distanceSquared($v3) < $pos1->distanceSquared($vector))){
-			$vector = $v3;
-		}
+			if($v4 !== null and ($vector === null or $pos1->distanceSquared($v4) < $pos1->distanceSquared($vector))){
+				$vector = $v4;
+			}
 
-		if($v4 !== null and ($vector === null or $pos1->distanceSquared($v4) < $pos1->distanceSquared($vector))){
-			$vector = $v4;
-		}
+			if($v5 !== null and ($vector === null or $pos1->distanceSquared($v5) < $pos1->distanceSquared($vector))){
+				$vector = $v5;
+			}
 
-		if($v5 !== null and ($vector === null or $pos1->distanceSquared($v5) < $pos1->distanceSquared($vector))){
-			$vector = $v5;
-		}
+			if($v6 !== null and ($vector === null or $pos1->distanceSquared($v6) < $pos1->distanceSquared($vector))){
+				$vector = $v6;
+			}
 
-		if($v6 !== null and ($vector === null or $pos1->distanceSquared($v6) < $pos1->distanceSquared($vector))){
-			$vector = $v6;
-		}
+			if($vector === null){
+				continue;
+			}
 
-		if($vector === null){
-			return null;
-		}
+			$f = -1;
+
+			if($vector === $v1){
+				$f = 4;
+			}elseif($vector === $v2){
+				$f = 5;
+			}elseif($vector === $v3){
+				$f = 0;
+			}elseif($vector === $v4){
+				$f = 1;
+			}elseif($vector === $v5){
+				$f = 2;
+			}elseif($vector === $v6){
+				$f = 3;
+			}
 
-		$f = -1;
-
-		if($vector === $v1){
-			$f = 4;
-		}elseif($vector === $v2){
-			$f = 5;
-		}elseif($vector === $v3){
-			$f = 0;
-		}elseif($vector === $v4){
-			$f = 1;
-		}elseif($vector === $v5){
-			$f = 2;
-		}elseif($vector === $v6){
-			$f = 3;
+			return MovingObjectPosition::fromBlock($this->x, $this->y, $this->z, $f, $vector->add($this->x, $this->y, $this->z));
 		}
 
-		return MovingObjectPosition::fromBlock($this->x, $this->y, $this->z, $f, $vector->add($this->x, $this->y, $this->z));
+		return null;
```

Comment:
The changed logic in this function is not correct - it ought to return the ray-traced point closest to the starting position.